src/java.sql/share/classes/java/sql/ResultSetMetaData.java

```diff
@@ -135,7 +136,7 @@
      *          of the designated column
      * @exception SQLException if a database access error occurs
      */
-    int getColumnDisplaySize(int column) throws SQLException;
+    @NonNegative int getColumnDisplaySize(@Positive int column) throws SQLException;
```

Comment:
I found multiple different implementations of this interface that sometimes return 0.  Are you sure that all those implementations are incorrect?  What are you basing this `@NonNegative` annotation on?

Reply:
`@NonNegative`'s documentation says "The annotated expression evaluates to an integer greater than or equal to 0. " which means that the return value includes 0. Did I get the usage of the annotation wrong?

Reply:
Sorry, I switched around the two annotations on this line.  I apologize for my mistake.